lib/p2p/NodeList.ts

```diff
@@ -256,6 +274,22 @@ class NodeList extends EventEmitter {
     return false;
   }
 
+  public incrementConsecutiveConnFailures = async (nodePubKey: string) => {
+    const node = this.nodes.get(nodePubKey);
+    if (node) {
```

Comment:
Should we maybe throw an error here (and in the reset method below) if the node does not exist?

Reply:
Other methods in this class are more graceful. I'm not sure if there's a good reason for that.
But anyway, I've changed the 2 methods i've added to use `assert`.

Reply:
Actually it's possible that the node entry wouldn't exist, if we never had a successful connection/session before.

Reply:
OK, I believe that's right actually, good to leave as is.